packages/edit-post/src/components/header/header-toolbar/index.js

```diff
@@ -73,6 +76,10 @@ function HeaderToolbar() {
 	/* translators: accessibility text for the editor toolbar */
 	const toolbarAriaLabel = __( 'Document tools' );
 
+	const toggleListView = useCallback(
+		() => setIsListViewOpened( ( isOpen ) => ! isOpen ),
+		[]
```

Comment:
Note that useState setters don’t update on each render https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often